runtime/lib/ttnn/operations/eltwise/where/where.cpp

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "where.h"
+#include "tt/runtime/detail/ttnn.h"
+#include "tt/runtime/ttnn/operations/utils.h"
+
+namespace tt::runtime::ttnn::operations::where {
+static void
+runWhereOp(::tt::target::ttnn::WhereOp const *op, ProgramTensorPool &tensorPool,
+           std::function<::ttnn::Tensor(
```

Comment:
Since `where` op is treated uniquely from other ternary ops, there is no need to use lambda here, just move those arguments from lambda to the `runWhereOp` function directly. In case in the future if we want to invoke `ttnn:where` with a different set of arguments, we will just have a `runWhereOp` with a different signature, similar to how Metal does with different `invoke` functions (https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/eltwise/ternary/where.cpp).